util/stmtsummary/v2/stmtsummary.go

```diff
@@ -309,6 +310,15 @@ func (s *StmtSummary) Close() {
 		s.closeWg.Wait()
 	}
 	s.closed.Store(true)
+	s.flush()
+}
+
+func (s *StmtSummary) flush() {
+	s.windowLock.Lock()
+	if s.window.lru.Size() > 0 {
+		s.storage.persist(s.window, timeNow())
```

Comment:
Since zap is asynchronous, I think we still need to explicitly call `logger.Sync` here.

The following method can be added:
```go
func (s *stmtLogStorage) sync() error {
	return s.logger.Sync()
}
```

And then:
```go
// ...
		s.storage.persist(s.window, timeNow())
		_ = s.storage.sync() // I'm not sure if this error is still loggable.
```

Reply:
Yes, it can log since the background logger will work until the last moment